schema/json.go

```diff
@@ -51,17 +51,45 @@ func (dst *JSON) Set(src interface{}) error {
 
 	switch value := src.(type) {
 	case string:
+		if value == "" {
+			*dst = JSON{Bytes: []byte(""), Status: Null}
+			return nil
+		}
+		// validate this is a valid json
+		err := json.Unmarshal([]byte(value), &struct{}{})
+		if err != nil {
+			return err
+		}
 		*dst = JSON{Bytes: []byte(value), Status: Present}
 	case *string:
 		if value == nil {
 			*dst = JSON{Status: Null}
```

Comment:
It's going out of the switch case so it's fine. Not the most consistent code because it's started with `switch case` the original code from `pgtype` and now we have more cases where I had to return. Not sure if better to change everything to return.